src-tauri/src/aux_functions/adj_matrix.rs

```diff
@@ -0,0 +1,104 @@
+use crate::graph::graph_ds::Graph;
+use std::collections::HashMap;
+
+/// Converts a graph to an adjacency matrix.
+///
+/// # Arguments
+///
+/// * `graph` - a graph
+///
+/// # Returns
+///
+/// * `Vec<Vec<f64>>` - an adjacency matrix
+pub fn convert_to_adj_matrix(graph: &Graph) -> Vec<Vec<f64>> {
+    let mut adj_matrix: Vec<Vec<f64>> = Vec::new();
+    let nodes = graph.get_nodes();
+    let edges = graph.get_edges();
+
+    let mut ordering_counter = 0 as usize;
+    let mut node_id_to_int = HashMap::new();
+
+    for node in nodes.clone() {
```

Comment:
Why are we cloning this?

Reply:
Because we are borrowing nodes later on. But we can also use &nodes here instead of cloning. I fixed this and it will be included in my next pr.